### PR TITLE
fixes #23 - Improve Liveness and Readiness checks

### PR DESCRIPTION
**Pull request by patilsuraj767**
With this PR in status API `/api/subscriptions/v1/status/` we will be checking if all the report processors/Threads are alive. 
If any of the report processor thread is exited then API will return 500 status code and liveness probe will get triggered.

### CODE REVIEW

File: yupana/api/status/view.py [hunk 1]
```diff
@@ -78,4 +78,9 @@ def status(request):
     serializer = StatusSerializer(status_info)
     server_info = serializer.data
     server_info['server_address'] = request.META.get('HTTP_HOST', 'localhost')
+    processor = list_name_of_processors()
```
Comment (gnurag) [inline yupana/api/status/view.py:81]: Perhaps to avoid confusion, the name should be `processors` since it returns a list?

**Review by gnurag — COMMENTED** [review]
Cosmetic suggestions.

File: yupana/api/status/view.py [hunk 2]
```diff
@@ -78,4 +84,11 @@ def status(request):
     serializer = StatusSerializer(status_info)
     server_info = serializer.data
     server_info['server_address'] = request.META.get('HTTP_HOST', 'localhost')
+    processors = list_name_of_processors()
```
Comment (kgaikwad) [inline yupana/api/status/view.py:87]:
```suggestion
    total_processors_names = list_name_of_processors()
```

File: yupana/api/status/view.py [hunk 3]
```diff
@@ -78,4 +84,11 @@ def status(request):
...
+    threads = list_name_of_active_threads()
```
Comment (kgaikwad) [inline yupana/api/status/view.py:88]:
```suggestion
    active_threads_names = list_name_of_active_threads()
```

File: yupana/api/status/view.py [hunk 4]
```diff
@@ -78,4 +84,11 @@ def status(request):
...
+    if not all(item in threads for item in processors):
+        dead_threads = set(processors).difference(threads)
```
Comment (kgaikwad) [inline yupana/api/status/view.py:90]:
```suggestion
        dead_processors = set(total_processors_names).difference(active_threads_names)
```

File: yupana/api/status/view.py [hunk 5]
```diff
@@ -78,4 +84,11 @@ def status(request):
...
+        LOG.error(format_message('SERVICE STATUS', 'Dead processors - %s' % dead_threads))
```
Comment (kgaikwad) [inline yupana/api/status/view.py:91]:
```suggestion
        LOG.error(format_message('SERVICE STATUS', 'Dead processors - %s' % dead_processors))
```

**Review by kgaikwad — CHANGES_REQUESTED** [review]
@patilsuraj767, LGTM :+1: 

Tested changes on local setup. 
When I stopped the processor's producer and respective loop, API `/api/subscriptions/v1/status/` returns 500 otherwise it is returning 200 OK.

Requested just small cosmetic changes.   Also, perform rebase.

**Review by kgaikwad — APPROVED** [review]
Thanks @patilsuraj767!

**Pull request merged**